src/spec-lang/tc/typecheck.ts

```diff
@@ -1500,6 +1516,86 @@ function matchArguments(
     return true;
 }
 
+/**
+ * We check the following in forall (uint t in [a ... b]) e(t):
+ *   - Type of t should be numeric
+ *   - a and b should be numeric and be castable to the type of t
+ *   - type of e is boolean
+ *   - return value of this expression is boolean
+ *   - t is defined in e(t).
+ */
+export function tcForAll(expr: SForAll, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode {
+    if (!(expr.iteratorType instanceof IntType)) {
+        throw new SWrongType(
+            `The expected type for ${expr.iteratorVariable.pp()} is numeric and not ${
+                expr.iteratorType
```

Comment:
expr.iteratorType needs `pp` here.

Reply:
I think we can remove this part, since parser checks this.